icefoganalytics/elcc-data-management
# ElCC-18: URL Upgrades and Model Additions To Handle Employee Benefits

Relates to:
- https://yg-hpw.atlassian.net/browse/ELCC-18

# Context

The Employee Benefits section math is set up the same as the rest of the spreadsheet, but it really should be quite different.

Currently it does the A x B = C math horizontally, and then adding up the totals vertically like everywhere else:
![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/7e93295e-b273-46dc-bf9d-6facc3bf33b5)

It should be just data fields for Gross, EE Cost and ER Cost, with the 8% calculating automatically, and then the Section Total being the lesser of 8% compared to Employer Cost.
![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/8692eedf-80f2-44e3-9f45-d3be43173fe4)

# Implementation

Pre-work cleanup before adding Employee Benefits UI.
Adds fiscal year slug as url param, to make it easier to lookup various data in dashboard tabs. e.g. http://localhost:8080/child-care-centres/1/2023-24/summary/reconciliation or http://localhost:8080/child-care-centres/1/2023-24/worksheets/september
Adds fiscal period model for to provide naming and shared location for per-month data.
Adds employee benefits model; the current one on http://localhost:8080/child-care-centres/1/2023-24/worksheets/april does not have enough flexibility to handle the requirements.

# Screenshots

![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/e93c1a32-dfbb-49b5-bb74-fdfb3ef1c7bb)

![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/14866ec4-ed22-4d6a-bd23-1175e81b1fb4)

![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/4839faa3-fb6b-44b7-9993-77466a0fe94b)

# Testing Instructions

1. Boot the app via `dev up`
2. Check that the app complies, and that you can log in at http://localhost:8080.
3. Click on the "Education is Currently Funding" widget to go to the list of centers page.
4. Click on the first child care center to open the "Child Care Centre Details" panel.
5. Click on the "View details" button in the bottom right corner.
6. Check that you are redirected to http://localhost:8080/child-care-centres/1/2023-24/summary/reconciliation; the current fiscal year.
7. Click through all the various tabs and check that they render correctly.
8. Go to a month like http://localhost:8080/child-care-centres/1/2023-24/worksheets/august and hard refresh the page.
9. Check that the content loads correctly with the appropriate year.
10. Switch the "Fiscal year" via the top right selector, and check that it shows whether there are worksheets for the year or not correctly.
